jerry-core/jerry.c

```diff
@@ -1991,10 +1992,7 @@ jerry_snapshot_set_offsets (uint8_t *buffer_p, /**< buffer */
         const_literal_end = args_p->const_literal_end;
       }
 
-      for (uint32_t i = 0; i < register_end; i++)
-      {
-        literal_start_p[i] = MEM_CP_NULL;
-      }
+      register_clear_start = 0;
```

Comment:
As always, I'd suggest moving declaration as close to initialization as possible. Here, it is possible to do them in one step. (C99 allows declaration later than at the beginning of a block, and we use it often.)